cmd/scaffold.go

```diff
@@ -204,6 +205,16 @@ func (fnb *FlowNodeBuilder) enqueueNetworkInit() {
 			return nil, fmt.Errorf("could not generate libp2p node factory: %w", err)
 		}
 
+		if fnb.Unstaked {
+			fnb.MsgValidators = []network.MessageValidator{
+				// filter out messages sent by this node itself
+				validator.NewSenderValidator(fnb.Me.NodeID()),
+				// but retain all the 1-k messages even if they are not intended for this node
```

Comment:
Can you explain this part a little more? Why do we filter out messages sent by this node itself, and what is the significance of 1-k?

Reply:
sure..currently, when you send a 1-k message (in other words, publish a message on a topic), the network layers seems to also loopback the message to the sender if the sender has also subscribed to the topic. This is not desired and hence a validator was added to filter out such messages which have the Origin field in the message equal to the node ID of the node (loop back messages).

Also, when you send a 1-k message, the sender has to specify a list of TargetIDs which indicate the nodes for which the message is intended. This is generally a subset of nodes subscribed on the topic. e.g. If a collection node sends a tx to collection node 1, 2 and 3, it will be received by ALL collection nodes in the system since they are all subscribed to the send-transaction channel. The target validator makes sure that only the intended targets receive the message and the other nodes just drop it.

Reply:
I guess my question is, why does the behavior of the staked vs unstaked node need to be different?

Reply:
@vishalchangrani does it simplify the management of the current unstaked access node set if unstaked receivers can pick up messages destined for an obsolete ID?